### ISSUE TITLE
Javadoc is missing a lot of information

### ISSUE THREAD

**Issue by joserobjr** [Kotlin/dokka#160]
With this simple test class:
```java
package test;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class Test
{
    @NotNull
    public static String test(@Nullable String test) {
        if(test == null)
            return "null";
        return test.toLowerCase();
    }
}
```
Dokka is generating an incomplete documentation page for it in javadoc mode.

It is missing the "extends Object" information, the "Methods inherited from class java.lang.Object", the default constructor "Test()", the method annotations and the method parameter.

---

## Page generated by dokka
![image](https://cloud.githubusercontent.com/assets/3679022/26227309/fe5830e2-3c07-11e7-85c2-dbb484ef2234.png)

---

## Page generated by javadoc
![image](https://cloud.githubusercontent.com/assets/3679022/26227296/e4c26d82-3c07-11e7-9857-6fd8dd2186af.png)

---

The same is applied to Kotkin classes but it is missing less information then Java classes.

Example Kotlin class:
```kotlin
class TestKotlin {
    fun test(test: String?): String {
        if (test == null)
            return "null"
        return test.toLowerCase()
    }
```
---

## Page generated by dokka
![image](https://cloud.githubusercontent.com/assets/3679022/26227494/495d1e6c-3c09-11e7-89a9-607e612c4113.png)

---

The most critical issues are the arguments missing in method information, the default constructor being omitted and the annotations which contains the `@Documented` decoration not being documented as they should.

**Comment by JJK3** [comment]
It looks like javadocs are still missing the method summary, has there been any progress on this? I'm willing to look into this and try to fix it, I just don't want to put needless effort into it if someone else is working on it.

**Comment by kamilok1965** [comment]
Yes and no. We'd really appreciate your help with this, @JJK3, just not now, because we're refactoring dokka quite extensively. We're changing the whole internal model, so fixing this javadoc format would be a waste of time, because it will have to be rewritten from scratch before the next big release (0.11.0).

**Comment by AfzalivE** [comment]
@kamildoleglo Do you mean that you need someone to implement `CommonLanguageService` in `JavaLanguageService`?

**Comment by Kordyjan** [comment]
As far as I know, this is fixed in new javadoc that was added in 1.4.0
